### ISSUE TITLE
Make time not required on deadline date field

### ISSUE THREAD

**Issue by pyrello** [uiowa/uiowa01#194]
Deadlines need to be able to function with or without a time specified.

#122

**Comment by mark-bennett-uiowa** [comment]
We have a couple of options here:
(1) Since the date and time is combined into one field, we would have to remove the Required flag on the entire field. The other content types do require the date/time, so for "deadlines" I assumed they would be required.
(2) separate the date and time into two separate fields. Require the date, do not require the time.

I am not thrilled about separating the field into two fields, so if anything I would vote for option 1. Thoughts?

**Comment by pyrello** [comment]
Still reading through this issue thread, but thought I would share it because it addresses the state of play regarding the "all day" feature that used to be standard on drupal date fields. https://www.drupal.org/project/drupal/issues/2734255

Here is a module seeking to address this issue that is mentioned in the above thread: https://www.drupal.org/project/date_all_day

I am in agreement that it would be better not to split the field, but I'm not satisfied with making the field optional, as the date part of the field really is not optional with a deadline type (at least as I need it to be implemented on the Graduate College site). So I think we are looking at an additional module, a patch, or some custom code to make this work.

**Comment by mark-bennett-uiowa** [comment]
I have been testing out the date_all_day module. Unfortunately, this does not give you the "All Day" checkbox like the D7 version did. You choose whether the field is one of the following types:
![Screen Shot 2019-11-12 at 3 48 10 PM](https://user-images.githubusercontent.com/3067017/68713733-21ca5c80-0564-11ea-85a2-fec5bb3bb0a4.png)
What this does is force you to choose whether you will ever allow times for this field. By checking "All Day", you prevent users from ever entering times:
![Screen Shot 2019-11-12 at 3 48 30 PM](https://user-images.githubusercontent.com/3067017/68713848-53dbbe80-0564-11ea-858f-a4ad98282762.png)

Once this module allows you to enter dates/times and provides you with an "All Day" checkbox that hides the time fields, then I think we could use this. Until then we need to find another solution.

**Comment by pyrello** [comment]
#sadface

**Comment by pyrello** [comment]
Here is another issue that is mentioned on the date_all_day module page: https://www.drupal.org/project/drupal/issues/2794481

@mark-bennett-uiowa I'm putting this here in case you are interested in reviewing it now, or for me to review later.

**Comment by mark-bennett-uiowa** [comment]
Yes, I saw that too. I heard they are possibly making end dates optional in core shortly. It looks like this is a wait and see also. Last updates were only 2-3 weeks ago, so hopefully something happens quick on this.

**Comment by pyrello** [comment]
If there is a patch that works, we could use it in the meantime. That issue is currently slated for 8.9.x, so it won't be released until next June when 9.0.0 is released.

**Comment by pyrello** [comment]
It looks like the patch to start testing against would be the one in this comment: https://www.drupal.org/project/drupal/issues/2794481#comment-12613817

There is another patch after that point, but it didn't look like it had been reviewed yet, nor was it passing the automated tests.

**Comment by pyrello** [comment]
It is crazy how unsettled the date/time handling landscape is as of 8.8.x, considering the next version of Drupal will be 9.0.x (8.9.x). I just came across this module: https://www.drupal.org/project/optional_end_date

It appears to be a module to bridge to the intended functionality in the next version of core. I'm going to try it out.
